code/game/objects/items/devices/uplinks.dm

```diff
@@ -172,6 +182,18 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 		show_description = text2num(href_list["show_desc"])
 		interact(usr)
 
+	else if(href_list["get_tc"])
+		if (uses <= 0)
+			return
+		var/amount = input("How many telecrystals do you wish to withdraw?:", "Extract telecrystals", null) as num
+		if (amount > uses)
+			return
+		uses -= amount
+		var/obj/item/stack/sheet/mineral/telecrystal/refined/R = new /obj/item/stack/sheet/mineral/telecrystal/refined(usr, amount)
+		var/mob/living/carbon/human/H = usr
+		H.put_in_hands(R)
```

Comment:
give a feedback message telling them how many they withdrew (useful to have this in chat in case you mistype without realizing it, you can scroll up and see "oh shit I hit 8 and instead of 7, that's why")